src/main/java/org/apache/commons/imaging/formats/tiff/TiffImageParser.java

```diff
@@ -870,9 +862,47 @@ TiffRasterData getFloatingPointRasterData(
             }
         }
 
-        if (predictor == TiffTagConstants.PREDICTOR_VALUE_HORIZONTAL_DIFFERENCING) {
-            throw new ImageReadException(
-                "TIFF floating-point data uses unsupported horizontal-differencing predictor");
+        if (sSampleFmt[0] == TiffTagConstants.SAMPLE_FORMAT_VALUE_IEEE_FLOATING_POINT) {
```

Comment:
I think `sSampleFmt` is returned from a call above this line,

```java
directory.getFieldValue(
            TiffTagConstants.TIFF_TAG_SAMPLE_FORMAT, true);

```

which may return `null`. So I think we have to wrap this `if/else` in a `if (sSampleFmt != null)`...

Reply:
done